### PR TITLE
feat: anti csrf changed to accept a function

### PR DESCRIPTION
**Pull request by iresharma**
## Summary of change

Convert `anti_csrf` into a function as part of app-info-refactor

## Related issues

-   Link to issue1 here
-   Link to issue1 here

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work. Bonus points for screenshots and videos!)

## Documentation changes

(If relevant, please create a PR in our [docs repo](https://github.com/supertokens/docs), or create a checklist here highlighting the necessary changes)

## Checklist for important updates

-   [ ] Changelog has been updated
-   [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `supertokens_python/constants.py`
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `setup.py`
    -   In `supertokens_python/constants.py`
-   [ ] Had installed and ran the pre-commit hook
-   [ ] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.
-   [ ] If have added a new web framework, update the `supertokens_python/utils.py` file to include that in the `FRAMEWORKS` variable
-   [ ] If added a new recipe that has a User type with extra info, then be sure to change the User type in supertokens_python/types.py
-   [ ] Make sure that `syncio` / `asyncio` functions are consistent.
 
## Remaining TODOs for this PR

-   [ ] Item1
-   [ ] Item2

### CODE REVIEW

File: supertokens_python/recipe/session/interfaces.py [hunk 1]
```diff
@@ -140,6 +140,7 @@ async def create_new_session(
         access_token_payload: Optional[Dict[str, Any]],
         session_data_in_database: Optional[Dict[str, Any]],
         disable_anti_csrf: Optional[bool],
+        anti_csrf: str,
```
Comment (rishabhpoddar) [inline supertokens_python/recipe/session/interfaces.py:143]:
```suggestion
```

File: supertokens_python/recipe/session/interfaces.py [hunk 2]
```diff
@@ -157,6 +158,7 @@ def get_global_claim_validators(
     async def get_session(
         self,
         access_token: Optional[str],
+        anti_csrf: str,
```
Comment (rishabhpoddar) [inline supertokens_python/recipe/session/interfaces.py:161]:
```suggestion
```

File: supertokens_python/recipe/session/interfaces.py [hunk 3]
```diff
@@ -197,6 +199,7 @@ async def refresh_session(
         refresh_token: str,
         anti_csrf_token: Optional[str],
         disable_anti_csrf: bool,
+        anti_csrf: str,
```
Comment (rishabhpoddar) [inline supertokens_python/recipe/session/interfaces.py:202]:
```suggestion
```

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: supertokens_python/recipe/session/syncio/__init__.py [hunk 4]
```diff
@@ -123,6 +123,7 @@ def get_session_without_request_response(
             MaybeAwaitable[List[SessionClaimValidator]],
         ]
     ] = None,
+    anti_csrf: Union[str, None] = None,
```
Comment (nkshah2) [inline supertokens_python/recipe/session/syncio/__init__.py:126]: Is this required here? Its been removed from everywhere else

File: supertokens_python/recipe/session/session_request_functions.py [hunk 5]
```diff
@@ -271,6 +273,8 @@ async def create_new_session_in_request(
         )
 
     disable_anti_csrf = output_transfer_method == "header"
```
Comment (nkshah2) [inline supertokens_python/recipe/session/session_request_functions.py:275]: We should rename this to something more accurate

File: supertokens_python/recipe/session/asyncio/__init__.py [hunk 6]
```diff
@@ -319,6 +319,7 @@ async def get_session_without_request_response(
             MaybeAwaitable[List[SessionClaimValidator]],
         ]
     ] = None,
+    anti_csrf: Union[str, None] = None,
```
Comment (nkshah2) [inline supertokens_python/recipe/session/asyncio/__init__.py:322]: This is not required

File: supertokens_python/recipe/session/recipe_implementation.py [hunk 7]
```diff
@@ -278,21 +275,15 @@ async def refresh_session(
...
             )
 
         log_debug_message("refreshSession: Started")
 
         response = await session_functions.refresh_session(
-            self,
-            refresh_token,
-            anti_csrf_token,
-            disable_anti_csrf,
+            self, refresh_token, anti_csrf_token, disable_anti_csrf
```
Comment (nkshah2) [inline supertokens_python/recipe/session/recipe_implementation.py:286]: Why did the formatting here change when the code hasnt?

**Review by nkshah2 — CHANGES_REQUESTED** [review]

**Review by nkshah2 — APPROVED** [review]

**Review by nkshah2 — APPROVED** [review]

**Review by nkshah2 — CHANGES_REQUESTED** [review]

**Review by nkshah2 — APPROVED** [review]

**Pull request merged**